In [2]:
# pic the best performing data from the datasets for each game.
import pandas as pd

path_results: str = '../../benchmark_results/results.csv'
df = pd.read_csv(path_results)

In [7]:
df_rev = df[df['Unnamed: 0'].str.contains('REV')] 

In [8]:
df_rev['Unnamed: 0'].unique()

array(['llama3.1-sft-e1-_E1_D10001_REV-t0.0--llama3.1-sft-e1-_E1_D10001_REV-t0.0',
       'llama3.1-sft-e1-_E1_D20001_REV-t0.0--llama3.1-sft-e1-_E1_D20001_REV-t0.0',
       'llama3.1-sft-e1-_E1_D20002_REV-t0.0--llama3.1-sft-e1-_E1_D20002_REV-t0.0',
       'llama3.1-sft-e1-_E1_D20003_REV-t0.0--llama3.1-sft-e1-_E1_D20003_REV-t0.0',
       'llama3.1-sft-e1-_E1_D30002_REV-t0.0--llama3.1-sft-e1-_E1_D30002_REV-t0.0',
       'llama3.1-sft-e1-_E1_D30003_REV-t0.0--llama3.1-sft-e1-_E1_D30003_REV-t0.0',
       'llama3.1-sft-e1-_E1_D30004_REV-t0.0--llama3.1-sft-e1-_E1_D30004_REV-t0.0',
       'llama3.1-sft-e1-_E1_D50001_REV-t0.0--llama3.1-sft-e1-_E1_D50001_REV-t0.0',
       'llama3.1-sft-e1-_E1_D50002_REV-t0.0--llama3.1-sft-e1-_E1_D50002_REV-t0.0',
       'llama3.1-sft-e1-_E1_D50003_REV-t0.0--llama3.1-sft-e1-_E1_D50003_REV-t0.0',
       'llama3.1-sft-e1-_E1_D50004_REV-t0.0--llama3.1-sft-e1-_E1_D50004_REV-t0.0',
       'llama3.1-sft-e1-_E1_D50005_REV-t0.0--llama3.1-sft-e1-_E1_D50005_REV-t0.0',
    

In [13]:
columns_to_keep = ['Unnamed: 0', '-, clemscore', 'all, Average % Played',
                   'all, Average Quality Score', 'imagegame, % Played',
                   'imagegame, Quality Score',
                   'privateshared, % Played', 'privateshared, Quality Score', 'referencegame, % Played',
                   'referencegame, Quality Score', 
                   'taboo, % Played', 'taboo, Quality Score', 
                   'wordle, % Played', 'wordle, Quality Score', 'wordle_withclue, % Played',
                   'wordle_withclue, Quality Score', 'wordle_withcritic, % Played',
                   'wordle_withcritic, Quality Score']

In [16]:
df_c = df_rev[columns_to_keep]

In [18]:
df_c.head(25)

Unnamed: 0  -, clemscore  \
6   llama3.1-sft-e1-_E1_D10001_REV-t0.0--llama3.1-...         27.37   
9   llama3.1-sft-e1-_E1_D20001_REV-t0.0--llama3.1-...         32.31   
12  llama3.1-sft-e1-_E1_D20002_REV-t0.0--llama3.1-...         25.68   
14  llama3.1-sft-e1-_E1_D20003_REV-t0.0--llama3.1-...         26.94   
17  llama3.1-sft-e1-_E1_D30002_REV-t0.0--llama3.1-...         32.44   
19  llama3.1-sft-e1-_E1_D30003_REV-t0.0--llama3.1-...         34.32   
21  llama3.1-sft-e1-_E1_D30004_REV-t0.0--llama3.1-...         33.75   
22  llama3.1-sft-e1-_E1_D50001_REV-t0.0--llama3.1-...         27.89   
23  llama3.1-sft-e1-_E1_D50002_REV-t0.0--llama3.1-...         24.63   
24  llama3.1-sft-e1-_E1_D50003_REV-t0.0--llama3.1-...         15.50   
25  llama3.1-sft-e1-_E1_D50004_REV-t0.0--llama3.1-...         27.72   
26  llama3.1-sft-e1-_E1_D50005_REV-t0.0--llama3.1-...         27.10   
27  llama3.1-sft-e1-_E3_D30001_REV-t0.0--llama3.1-...         33.07   
29  llama3.1-sft-e1-_E1_D50006_REV-t0.0--llama3.1-...         14.72   
30  llama3.1-sft-e1-_E1_D50007_REV-t0.0--llama3.1-...         16.74   
34  llama3.1-sft-e1-_E3_D30001_REV-t0.0--llama3.1-...         33.07   

    all, Average % Played  all, Average Quality Score  imagegame, % Played  \
6                   92.86                       29.47                 90.0   
9                   90.60                       35.66                 97.5   
12                  88.57                       28.99                 90.0   
14                  91.67                       29.39                 95.0   
17                  95.11                       34.11                100.0   
19                  93.33                       36.77                100.0   
21                  97.44                       34.64                100.0   
22                  95.70                       29.14                 97.5   
23                  80.75                       30.50                 72.5   
24                  72.08                       21.50                  2.5   
25                  81.72                       33.92                 85.0   
26                  83.89                       32.30                100.0   
27                  95.14                       34.76                100.0   
29                  74.29                       19.82                 17.5   
30                  81.90                       20.44                 70.0   
34                  95.14                       34.76                100.0   

    imagegame, Quality Score  privateshared, % Played  \
6                      65.22                    100.0   
9                      72.41                    100.0   
12                     50.28                    100.0   
14                     54.42                    100.0   
17                     83.25                     94.0   
19                     73.25                    100.0   
21                     81.42                     98.0   
22                     41.95                    100.0   
23                     40.14                     42.0   
24                     33.00                     40.0   
25                     47.21                     32.0   
26                     84.58                     40.0   
27                     84.88                     96.0   
29                     36.71                     66.0   
30                     28.61                    100.0   
34                     84.88                     96.0   

    privateshared, Quality Score  referencegame, % Played  \
6                          30.34                   100.00   
9                          28.71                   100.00   
12                         30.56                   100.00   
14                         53.82                   100.00   
17                         12.30                   100.00   
19                         17.77                   100.00   
21                         27.36                   100.00   
22                         40.93                  

In [19]:
len(df_c)

16

### Best Scores
Imagegame: p 100: q 84.88 D30001

privateshared: p 100 q 53.82 D20003

referencegame: p 100 q 40.00 D20002

taboo: p 100 q 56.7 D30003

wordle: p 93.3 q 6.06 D30001

wordle_withclie  p 76.67, q 33.48 D20001

wordle_withcritic p 95.00 q 31.58 D30001


In [21]:
path_image_game: str = '../../data/training_data/D30001.csv'
path_private_shared: str = '../../data/training_data/D20003.csv'
path_reference_game: str = '../../data/training_data/D20002.csv'
path_taboo: str = '../../data/training_data/D30003.csv'
path_wordle: str = '../../data/training_data/D30001.csv'
path_wordle_withclue: str = '../../data/training_data/D20001.csv'
wordle_withcritic: str = '../../data/training_data/D30001.csv'

In [22]:
d30001 = pd.read_csv(path_image_game)
d20003 = pd.read_csv(path_private_shared)
d20002 = pd.read_csv(path_reference_game)
d30003 = pd.read_csv(path_taboo)
d20001 = pd.read_csv(path_wordle_withclue)

In [27]:
df30001 = d30001[d30001.game.isin(['imagegame', 'wordle', 'wordle_withcritic'])]
df20003 = d20003[d20003.game.isin(['privateshared'])]
df20002 = d20002[d20002.game.isin(['referencegame'])]
df20001 = d20001[d20001.game.isin(['wordle_withclue'])]
df30003 = d30003[d30003.game.isin(['taboo'])]

df = pd.concat([df30001, df20003, df20002, df20001, df30003])


In [28]:
df.game.unique()

array(['wordle_withcritic', 'imagegame', 'wordle', 'privateshared',
       'referencegame', 'wordle_withclue', 'taboo'], dtype=object)

In [33]:
df.to_csv('../../data/training_data/D8000X.csv',index=False)

game  game_id  \
22  wordle_withcritic        4   
23  wordle_withcritic        4   
36  wordle_withcritic        5   
37  wordle_withcritic        5   
51          imagegame       13   
52          imagegame       13   
53          imagegame       13   
54          imagegame       13   
55          imagegame       13   
56  wordle_withcritic        2   
57  wordle_withcritic        2   
58  wordle_withcritic        2   
59  wordle_withcritic        2   
60  wordle_withcritic        2   
61  wordle_withcritic        2   
62  wordle_withcritic        2   
63  wordle_withcritic        2   
67          imagegame       12   
68          imagegame       12   
69          imagegame       12   
70          imagegame       12   
71          imagegame       12   

                                                model benchmark_version  \
22  Nous-Hermes-2-Mixtral-8x7B-DPO-t0.0--Nous-Herm...              v1.0   
23  Nous-Hermes-2-Mixtral-8x7B-DPO-t0.0--Nous-Herm...              v1.0   
36                   gpt-4-0613-t0.0--gpt-4-0613-t0.0              v1.0   
37                   gpt-4-0613-t0.0--gpt-4-0613-t0.0              v1.0   
51                   gpt-4-0613-t0.0--gpt-4-0613-t0.0              v1.0   
52                   gpt-4-0613-t0.0--gpt-4-0613-t0.0              v1.0   
53                   gpt-4-0613-t0.0--gpt-4-0613-t0.0              v1.0   
54                   gpt-4-0613-t0.0--gpt-4-0613-t0.0              v1.0   
55                   gpt-4-0613-t0.0--gpt-4-0613-t0.0              v1.0   
56         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
57         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
58         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
59         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
60         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
61         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
62         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
63         vicuna-13b-v1.5-t0.0--vicuna-13b-v1.5-t0.0              v1.0   
67   gpt-4-1106-preview-t0.0--gpt-4-1106-preview-t0.0              v1.0   
68   gpt-4-1106-preview-t0.0--gpt-4-1106-preview-t0.0              v1.0   
69   gpt-4-1106-preview-t0.0--gpt-4-1106-preview-t0.0              v1.0   
70   gpt-4-1106-preview-t0.0--gpt-4-1106-preview-t0.0              v1.0   
71   gpt-4-1106-preview-t0.0--gpt-4-1106-preview-t0.0              v1.0   

                                   experiment     episode  Aborted  Lose  \
22     2_low_frequency_words_clue_with_critic   episode_3        0     0   
23     2_low_frequency_words_clue_with_critic   episode_3        0     0   
36  1_medium_frequency_words_clue_with_critic   episode_4        0     0   
37  1_medium_frequency_words_clue_with_critic   episode_4        0     0   
51                             1_random_grids  episode_13        0     0   
52                             1_random_grids  episode_13        0     0   
53                             1_random_grids  episode_13        0     0   
54                             1_random_grids  episode_13        0     0   
55                             1_random_grids  episode_13        0     0   
56     2_low_frequency_words_clue_with_critic   episode_1        0     0   
57     2_low_frequency_words_clue_with_critic   episode_1        0     0   
58     2_low_frequency_words_clue_with_critic   episode_1        0     0   
59     2_low_frequency_words_clue_with_critic   episode_1        0     0   
60     2_low_frequency_words_clue_with_critic   episode_1        0     0   
61     2_low_frequency_words_clue_with_critic   episode_1        0     0   
62     2_low_frequency_words_clue_with_critic   episode_1        0     0   
63     2_low_frequency_words_clue_with_critic   episode_1        0     0   
67                             1_random_grids  episode_12        0     0   
68                             1_random_grids  episode_12       

In [1]:
import pandas as pd

path_results: str = '../../data/processed/privateshared_old_processed.jsonl'
df = pd.read_json(path_results, lines=True)

In [3]:
df.chat.iloc[0]

[{'role': 'user',
  'content': 'You are a customer of a travel agency. Here is a description of the details of the travel you want to make:\n\nWHAT: Travel\nFROM: Cologne\nTO: Lisbon\nBY: Overnight bus\nCLASS: The most comfortable\nWHEN: Anytime next week\n\nThe travel agent does not know about it yet. \n\nQuestions from the travel agent will start with TRAVEL-AGENT. Please reply in the form: ANSWER: <some text>\n\nI will also ask you questions. These questions will start with ME: . Please answer these questions with: ASIDE: <some text>\n\nImportant: Give short, direct answers!\n\nLet us start.',
  'type': 'send message'},
 {'role': 'user',
  'content': 'ME:  Have you told the travel agent where you are leaving from? Please answer yes or no.',
  'type': 'probe question'},
 {'role': 'assistant', 'content': 'ASIDE: Yes', 'type': 'probe answer'},
 {'role': 'user',
  'content': 'ME:  Does the travel agent already know your destination? Please answer yes or no.',
  'type': 'probe question'}